aws-s3/src/main/scala/busymachines/pureharm/aws/s3/package.scala

```diff
@@ -52,6 +52,6 @@ package object s3 {
   object S3DownloadURL extends PhantomType[String]
   type S3DownloadURL = S3DownloadURL.Type
 
-  object S3KeyPrefix extends PhantomType[String]
-  type S3KeyPrefix = S3KeyPrefix.Type
+  object S3PathPrefix extends PhantomType[String]
```

Comment:
I think you can drop the `Prefix` part, isn't it just a path?

Also, wouldn't it actually be roughly the same as an S3FileKey? What happens when you list on a file, instead of a folder?

Reply:
It returns the file's descriptor. So yeah, it kinda works the same as the S3FileKey, but the semantic is different. What do you think, reuse the S3FileKey in this context or create another phantom type that works the same as it?

Reply:
S3Path should be fine, different semantics, different types to express them.